File: src/backend/SyllabusManager.Data/Models/Subjects/Lesson.cs
using SyllabusManager.Data.Enums.Subjects;
using System.Collections.Generic;

namespace SyllabusManager.Data.Models.Subjects
{
    public class Lesson : ModelBase
    {
        public LessonType LessonType { get; set; }
        public int HoursAtUniversity { get; set; }
        public int StudentWorkloadHours { get; set; }
        public FormOfCrediting FormOfCrediting { get; set; }
        public int Ects { get; set; }
        public int EctsinclPracticalClasses { get; set; }
        public int EctsinclDirectTeacherStudentContactClasses { get; set; }
        public bool IsFinal { get; set; }
        public bool IsScientific { get; set; }
        public bool IsGroup { get; set; }
        public List<ClassForm> ClassForms { get; set; }
    }
}

File: src/frontend/src/app/core/models/subject/lesson.ts
import { FormOfCrediting } from "../../enums/subject/form-of-crediting.enum";
import { LessonType } from "../../enums/subject/lesson-type.enum";
import { ModelBase } from "../model-base";
import { ClassForm } from "./class-form";

export class Lesson extends ModelBase {
    lessonType?: LessonType;
    hoursAtUniversity: number = 0;
    studentWorkloadHours: number = 0;
    formOfCrediting: FormOfCrediting = FormOfCrediting.CreditingWithGrade;
    ects: number = 0;
    ectsinclPracticalClasses: number = 0;
    ectsinclDirectTeacherStudentContactClasses: number = 0;
    isFinal: boolean = false;
    isScientific: boolean = false;
    isGroup: boolean = false;
    classForms: ClassForm[] = [];
}
